### ISSUE TITLE
Two dashes replaces with long dash even in code mode

### ISSUE THREAD

**Issue by aNNiMON** [telegramdesktop/tdesktop#4706]
### Steps to reproduce
1. Start typing something like this
```
`i--` is decrement
```
or this

    ```
    while (i-- > 0) {
    }
    ```

### Expected behaviour
No problem with two dashes `--`

### Actual behaviour
Two dashes `--` automatically replaces with long dash `—` 
```
`i—` is decrement
```

    ```
    while (i— > 0) {
    }
    ```
And it's annoying, `cause I need press <kbd>Backspace</kbd> whenever I type two dashes.

P.S. Try to type:
```
`----------------`
```
and feel the pain :smirk: 

### Configuration
**Operating system:** Windows

**Version of Telegram Desktop:** 1.2.21 alpha

**Used theme**: Any

**Comment by Aokromes** [comment]
#522

**Comment by aNNiMON** [comment]
#522 is for server replacement and it normally handles double dashes between <code>``</code>

Now the app replaces dashes in editbox immediately, ignoring code tags

**Comment by stek29** [comment]
There’s no “server replacement”, but I agree that it’s a different issue.

**Comment by Lonami** [comment]
![image](https://user-images.githubusercontent.com/6297805/41410912-51edfa9a-6fdb-11e8-9ee4-c344aa95fb66.png)

This issue should probably be closed now (at least 1.3.7 alpha works).
